File: Python/Add Digits.py
class Solution(object):
    def addDigits(self, num):
        """
        :type num: int
        :rtype: int
        """
        temp = 0
        while num > 0:
            temp += num % 10
            num /= 10
        if temp < 10:
            return temp
        else:
            return self.addDigits(temp)

File: cpp/Add Digits.cpp
class Solution {
public:
    int addDigits(int num) {
        int temp = num;
        while (temp > 9) {
            temp = 0;
            while (num > 0) {
                temp += num % 10;
                num /= 10;
            }
            num = temp;
        }
        return temp;
    }
};
